R/write.R

```diff
@@ -0,0 +1,364 @@
+
+  # Construct default supertibble sheet ----
+  if (include_toc_from_supertbl) {
+    supertbl_toc <- add_supertbl_toc(wb, supertbl, include_metadata, labelled, table_style, set_col_widths)
+  }
+
+  # Write all redcap_form_name to sheets ----
+  map(
+    sheet_vals,
+    \(x) wb$add_worksheet(sheet = x)
+  )
+
+  # Write all redcap_data to sheets ----
+  # Account for special case when a dataframe may have zero rows
+  # This causes an error on opening the Excel file.
+  # Instead, apply a row of auto-determined NA types.
+  for (i in seq_len(nrow(supertbl))) {
+    if (nrow(supertbl$redcap_data[[i]]) == 0) {
+      supertbl$redcap_data[[i]] <- supertbl$redcap_data[[i]][1, ]
+    }
+  }
+
+  map2(
+    supertbl$redcap_data,
+    sheet_vals,
+    \(x, y) wb$add_data_table(sheet = y, x = x,
+                              startRow = ifelse(labelled, 2, 1),
+                              tableStyle = table_style)
+  )
+
+  # Construct default metadata sheet ----
+  if (include_metadata) {
+    add_metadata_sheet(supertbl, wb, labelled, table_style, set_col_widths)
+  }
+
+  # Apply additional aesthetics ----
+  # Apply standard colwidth
+  map2(
+    supertbl$redcap_data,
+    sheet_vals,
+    \(x, y) wb$set_col_widths(sheet = y,
+                              cols = seq_len(ncol(x)),
+                              widths = set_col_widths)
+  )
+
+  # Add labelled features ----
+  if (labelled) {
+    add_labelled_xlsx_features(
+      supertbl,
+      wb,
+      sheet_vals,
+      include_toc_from_supertbl,
+      include_metadata,
+      supertbl_toc
+    )
+  }
+
+  # Export workbook object ----
+  wb$save(path = file, overwrite = TRUE)
+}
+
+#' @title Add labelled features to write_redcap_xlsx
+#'
+#' @description
+#' Helper function to support `labelled` aesthetics to XLSX supertibble output
+#'
+#' @param supertbl a supertibble generated using `read_redcap()`
+#' @param wb An `openxlsx2` workbook object
+#' @param sheet_vals Helper argument passed from `write_redcap_xlsx` to
+#' determine and assign sheet values.
+#' @param include_toc_from_supertbl Include a sheet capturing the supertibble output.
+#' Default `TRUE`.
+#' @param include_metadata Include a sheet capturing the combined output of the
+#' supertibble `redcap_metadata`. Default `TRUE`.
+#' @param supertbl_toc The table of contents supertibble defined in the parent
+#' function. Default `NULL`.
+#'
+#' @importFrom purrr map pluck
+#' @importFrom tidyr pivot_wider
+#' @importFrom dplyr select filter
+#'
+#' @keywords internal
+
+add_labelled_xlsx_features <- function(supertbl,
+                                       wb,
+                                       sheet_vals,
+                                       include_toc_from_supertbl = TRUE,
+                                       include_metadata = TRUE,
+                                       supertbl_toc = NULL) {
+
+  # Generate variable labels off of labelled dictionary objects ----
```

Comment:
Since we only `Suggest` `labelled` we need to add `check_installed()` or `is_installed()` before any user facing calls to `labelled`